java/Oleksii-Riabov/src/com/zoolatech/lecture2/tasks/_2/Robot.java

```diff
@@ -0,0 +1,174 @@
+package com.zoolatech.lecture2.tasks._2;
+
+public class Robot {
+    private final Room room;
+    Direction direction;
+    private int width;
+    private int height;
+
+    public Robot() {
+        this.room = new Room(1, 1);
+        this.direction = Direction.SOUTH;
+        this.width = 0;
+        this.height = 0;
+    }
+
+    public Robot(int roomWidth, int roomHeight, String direction, int robotsStartingWidth, int robotStartingHeight) {
+        this.room = new Room(roomWidth, roomHeight);
+        this.direction = Direction.valueOf(direction);
+        this.width = robotsStartingWidth;
+        this.height = robotStartingHeight;
+    }
+
+    public Room getRoom() {
+        return room;
+    }
+
+    public Direction getDirection() {
+        return direction;
+    }
+
+    public int getWidth() {
+        return width;
+    }
+
+    public int getHeight() {
+        return height;
+    }
+
+    public void warning() {
+        System.out.println("The Robot reach the end of a room and can't move forward, please change Robot's direction.");
+    }
+
+    public void checkLocation() {
+        if (this.width < 0 & this.height < 0) {
```

Comment:
In what situation is't possible that both width and height are < 0? Same for this condition `this.width > room.width & this.height > room.height`. I think we move only in one direction.

Reply:
Already remove this redundant if-staements.